[AS-4029] Adding new event to track address validation

daytavares opened the pull request:
The type of this PR is: **Feature**

<!-- Bugfix/Feature/Enhancement/Documentation -->

<!-- If applicable, write the Jira ticket number in square brackets e.g. `[CO-434]`
     The Jira integration will turn it into a clickable link for you. -->

This PR resolves [EMI-1285](https://artsyproduct.atlassian.net/browse/EMI-1285)

### Description

This PR adds a new event to Cohesion to track when an user clicks on the buttons on the address verification modals

More details [here](https://www.notion.so/artsy/Address-auto-completion-and-verification-d3a89ec48845408f9d3a2724ce4e09b8?pvs=4#a6d4cf8409504609bdfdb309cd930de3)

### PR Checklist (tick all before merging)

<!-- 💡 This checklist is experimental. #cohesion warmly welcomes any feedback about the list or how it impacts your workflow -->

- [ ] If I've added a new file to the tree I've exported it from the common [`index.ts`](https://github.com/artsy/cohesion/blob/main/src/Schema/Events/index.ts)
- [ ] I've added comments with examples for any new interfaces and ensured that they're in the docs
- [ ] No platform-specific terminology has been used outside of `click` and `tap` (platform is inferred by the DB storing events)


[EMI-1285]: https://artsyproduct.atlassian.net/browse/EMI-1285?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

daytavares commented:
@erikdstock do you have comments or can I merge it?

erikdstock reviewed (approved):
Looks good from here!
two notes:
- I'm assuming 'clicked a button' with the button content is preferable to multiple separate events for each button.
- do we need any special events for when the user sees the verification modal?

Assuming these points are good, merge when ready :+1:

daytavares commented:
@erikdstock good point! 

- Maybe I can rename the event to make clear it's the buttons.
- I would maybe use the label to identify which button. I forgot sharing [this](https://docs.google.com/spreadsheets/d/1c65vugkF2C4YuXwBXno8aqsBh4B2fTwxiLmgY_MhjWc/edit#gid=0).
- Yes! I was not sure if we needed a special event when the user sees the modal. I thought it was like a pageview. I'll create one then.

TMcMeans commented:
hey, im taking over for adding the address validation event tracking from @erikdstock, and had a follow up question about if there will also need tracking events for when the modal is closed along with when the user sees it?

TMcMeans reviewed (commented):
hey, taking over adding address validation event tracking to force after @erikdstock passed [this ticket](https://artsyproduct.atlassian.net/browse/EMI-1285) to me, and had a follow up question about if we will also need tracking events for when the modal is closed along with when the user sees it?

daytavares commented:
Hi @TMcMeans!

I changed again a bit... I got confused with Erik's comments.

What we want is...

- Tracking if the modal opened (for that I am using impression event)
- If the modal opened, what are the clicks (for that I am using click event)

I added [here](https://docs.google.com/spreadsheets/d/1c65vugkF2C4YuXwBXno8aqsBh4B2fTwxiLmgY_MhjWc/edit#gid=0).

Let me know if it makes sense 😊

cc @louislecluse

daytavares commented:
added option selected 😊

artsyit commented:
<!-- GITHUB_RELEASE COMMENT: released -->
:rocket: PR was released in [`v4.134.0`](https://github.com/artsy/cohesion/releases/tag/v4.134.0) :rocket:

starsirius commented:
@daytavares Sorry if this is discussed but I have some followup questions:

- Maybe we can differentiate which modal is viewed by users, especially given one of the button copy is identical. This way we can analyze what gets triggered.
- Re-iterate @TMcMeans 's question above. Do you think we should track modal close, such as clicking the "X" button or outside the modal to dismiss it?

![Screen Shot 2023-07-19 at 3 46 28 PM](https://github.com/artsy/cohesion/assets/796573/e0ea1825-bd42-4a89-9744-56d375e7b82a)

TMcMeans commented:
@starsirius I pushed a pr with a small change to the event tracking schema for the modal types which should cover the distinction between modalTypes here: https://github.com/artsy/cohesion/pull/445

TMcMeans commented:
@daytavares would love your opinion on tracking the modal close- currently I can add it to three places which would be on each button press (edit or use this address) and on closing the modal by clicking the "x" button or away from the modal on the screen.

MounirDhahri commented on src/Schema/Events/Click.ts:
```diff
@@ -1607,3 +1607,32 @@ export interface ClickedOnDuplicateArtwork {
+ *    context_module: "OrdersShipping",
+ *    context_page_owner_type: "orders-shipping",
+ *    context_page_owner_id: "57e60c68-a198-431e-8a02-6ecb01e3a99b",
+ *    user_id: "61bcda16515b038ce5000104"
```
Hey, if this is going to be the logged-in user `user_id`, you probably don't need to set it since it's usually added by default to analytics events.
